course2/meeting10/main.cc

```diff
@@ -0,0 +1,54 @@
+#include <random>
+#include <algorithm>
+#include <vector>
+#include <list>
+#include <deque>
+#include <iostream>
+#include <chrono>
+#include <functional>
+
+struct RNG
+{
+	std::default_random_engine randEngine;
+	std::uniform_real_distribution<double> realDist{0, 1000};
+	double operator() () { return realDist(randEngine); }
+};
+
+void PrintTime(std::function<void()> func, const char* message)
```

Comment:
std::function is really cool, becuase it erases the type - allows you to squeeze all the different lambdas into one type.
But it also introduces overhead.

In this case, making the function a template would be better:
1. You don't need the type erasure in this case
2. In case of a benchmark, any overhead introduces noise into the result